File: redsheet-portal/src/app/database-models/request-log.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class RequestLog {
    constructor(requestLog?: RequestLog )  {
        let properties = [ "applicationModuleId", "authorizationHeader", "browserName", "clientIPAddress", "contentLength", "cookies", "parameters", "recordId", "requestLogId", "requestMethod", "requestTime", "responseStatusCode", "serviceUri", "totalDuration", "userId",];
        for (let property of properties)
            if (requestLog && requestLog[property])
                this[property] = requestLog[property];
    }
 
	applicationModuleId : number =   undefined;
 
	authorizationHeader : string =   undefined;
 
    @maxLength(200)
	browserName : string =   undefined;
 
    @maxLength(50)
	clientIPAddress : string =   undefined;
 
	contentLength : number =   undefined;
 
	cookies : string =   undefined;
 
    @required()
	parameters : string =   undefined;
 
	recordId : number =   undefined;
 
	requestLogId : number =   0 ;
 
    @required()
    @maxLength(10)
	requestMethod : string =   undefined;
 
    @required()
	requestTime : Date =   undefined;
 
    @range(1,2147483647)
	responseStatusCode : number =   undefined;
 
    @required()
    @maxLength(100)
	serviceUri : string =   undefined;
 
    @required()
	totalDuration : Date =   undefined;
 
	userId : number =   undefined;


}

File: RedSheet.DbEntities/Models/RequestLog.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("RequestLogs", Schema = "dbo")]
	[Serializable]
	public partial class RequestLog  
	{

	public Nullable<int> ApplicationModuleId { get; set; } 

	public string AuthorizationHeader { get; set; } 
 
	[MaxLength(200)]
	public string BrowserName { get; set; } 
 
	[MaxLength(50)]
	public string ClientIPAddress { get; set; } 

	public Nullable<int> ContentLength { get; set; } 

	public string Cookies { get; set; } 
 
	[Required]
	public string Parameters { get; set; } 

	public Nullable<int> RecordId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int RequestLogId { get; set; } 
 
	[Required] 
	[MaxLength(10)]
	public string RequestMethod { get; set; } 
 
	[Required]
	public System.DateTime RequestTime { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ResponseStatusCode { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string ServiceUri { get; set; } 
 
	[Required]
	public TimeSpan TotalDuration { get; set; } 

	public Nullable<int> UserId { get; set; } 

	public RequestLog()
	{
    }

    }
}

